server/src/main/java/io/deephaven/server/table/ops/TableServiceGrpcImpl.java

```diff
@@ -296,12 +305,116 @@ public void updateBy(UpdateByRequest request, StreamObserver<ExportedTableCreati
         oneShotOperationWrapper(BatchTableRequest.Operation.OpCase.UPDATE_BY, request, responseObserver);
     }
 
+    private Object getSeekValue(Literal literal, Class<?> dataType) {
+        if (literal.hasStringValue()) {
+            if (!String.class.isAssignableFrom(dataType)) {
+                throw GrpcUtil.statusRuntimeException(Code.INVALID_ARGUMENT,
+                        "Invalid String type for seek: " + dataType);
+            }
+            return literal.getStringValue();
+        } else if (literal.hasNanoTimeValue()) {
+            if (!DateTime.class.isAssignableFrom(dataType)) {
+                throw GrpcUtil.statusRuntimeException(Code.INVALID_ARGUMENT, "Invalid Date type for seek: " + dataType);
+            }
+            return new DateTime(literal.getNanoTimeValue());
+        } else if (literal.hasLongValue()) {
+            Long longValue = literal.getLongValue();
+            if (dataType == byte.class) {
+                return longValue.byteValue();
+            }
+            if (dataType == short.class) {
+                return longValue.shortValue();
+            }
+            if (dataType == int.class) {
+                return longValue.intValue();
+            }
+            if (dataType == long.class) {
+                return longValue;
+            }
+            if (dataType == float.class) {
+                return longValue.floatValue();
+            }
+            if (dataType == double.class) {
+                return longValue.doubleValue();
+            }
+            if (dataType == BigInteger.class) {
```

Comment:
If we expect these to work, we'll need BigIntegerWrapper/etc support too?

It seems odd to expect a long (i.e. likely from reading from one column of the table) to search a biginteger/bigdecimal field (which would be a different column).

Is this speculative, how we expect it might work in the future? Or does this work already and I misunderstand?

Reply:
True, I guess I should be converting from a String in the BigInteger/BigWrapper case (since Literal doesn't seem to have an option for either, best to pass as string and avoid truncation).
I'll remove the conversion from long/double.

Reply:
Side note - why do the methods `Literal.getLongValue` and `Literal.getDoubleValue` have the wrong return type? It's returning a boxed type but the signature is for the primitive type, e.g.:
```
  @java.lang.Override
  public long getLongValue() {
    if (valueCase_ == 4) {
      return (java.lang.Long) value_;
    }
    return 0L;
  }
```
Seems odd, and since I'm doing `Long longValue = literal.getLongValue()`, so it's getting autounboxed/boxed in that case?

Reply:
(Note that we're talking about the JVM implementation now, io.deephaven.proto.backplane.grpc.Literal is not used in the JS client via GWT)

Yes, Java will autounbox in cases like this. Since value_ is an Object, it can't be returned as a `long` primitive, it has to be cast to java.lang.Long, then the compiler will auto-unbox it to a long. Unless the value is null (which will not fail the cast), as that will throw a NullPointerException rather than return a value.